mungegithub/mungers/publisher.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/contrib/mungegithub/features"
+	"k8s.io/contrib/mungegithub/github"
+	"k8s.io/kubernetes/pkg/util/errors"
+)
+
+type Info struct {
+	Repo string
+	Dir  string
+}
+
+type PublisherMunger struct {
+	// base to all repos
+	baseDir      string
+	srcToDst     map[Info]Info
+	features     *features.Features
+	githubConfig *github.Config
+	script       string
+}
+
+func init() {
+	publisherMunger := &PublisherMunger{}
+	RegisterMungerOrDie(publisherMunger)
+}
+
+// Name is the name usable in --pr-mungers
+func (p *PublisherMunger) Name() string { return "publisher" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (p *PublisherMunger) RequiredFeatures() []string { return []string{features.RepoFeatureName} }
+
+// Initialize will initialize the munger
+func (p *PublisherMunger) Initialize(config *github.Config, features *features.Features) error {
+	p.baseDir = features.Repos.BaseDir
+	if len(p.baseDir) == 0 {
```

Comment:
We perform this check already when [initializing the gitrepo feature](https://github.com/foxish/contrib/blob/d0bbbe0db0c68ce2d7e7b47ebe21ef12391ed8f5/mungegithub/features/repo-updates.go#L195), so it may be redundant here.